### PR TITLE
[wip] Add logs for kubernetes events

### PR DESCRIPTION
**Pull request by sk593**
# Description

Adding logs to track pod failures during schedule func test runs

## Type of change

- This pull request fixes a bug in Radius and has an approved issue (issue link required).
- This pull request adds or changes features of Radius and has an approved issue (issue link required).
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: #issue_number

### Summary
🐽🛠

Enhance functional-test workflow to capture pod logs and events on failure. Add `id` to `publish-tf-recipes` step and two new steps to `.github/workflows/functional-test.yaml`.

> _When the recipes fail to publish_
> _We unleash the wrath of `pod logs`_
> _We capture the events of doom_
> _And upload them to the cloud of gloom_

### Walkthrough
* Add an `id` to the recipe publishing step to enable conditional execution of subsequent steps ([link](https://github.com/radius-project/radius/pull/6793/files?diff=unified&w=0#diff-c79f364a9293abaaa8595776b74674e24bec6287834e63ab8aa7aec6a42f0dbcR483))

### CODE REVIEW

File: .github/workflows/functional-test.yaml [hunk 1]
```diff
@@ -559,6 +560,30 @@ jobs:
             --subscription ${{ secrets.INTEGRATION_TEST_SUBSCRIPTION_ID }} \
             --name ${{ env.AZURE_TEST_RESOURCE_GROUP }} \
             --yes --verbose
+      # TODO add once tested: if: failure() && steps.publish-tf-recipes.outcome == 'failure'
```
Comment (rynowak) [inline .github/workflows/functional-test.yaml:563]: Would it be bad if we just always included these logs?

File: .github/workflows/functional-test.yaml [hunk 2]
```diff
@@ -559,6 +560,30 @@ jobs:
...
+        id: get-pod-logs
+        run: |
+          # Create pod-logs directory
+          mkdir -p recipes/pod-logs
+          # Get pod logs and save to file
+          namespace="radius-test-tf-module-server"
+          label="app.kubernetes.io/name=tf-module-server"
+          pod_names=($(kubectl get pods -l $label -n $namespace -o jsonpath='{.items[*].metadata.name}'))
+          for pod_name in "${pod_names[@]}"; do
+            kubectl logs $pod_name -n $namespace > recipes/pod-logs/${pod_name}.txt
```
Comment (ytimocin) [inline .github/workflows/functional-test.yaml:574]: We can add a log here like `echo "Pod ${pod_name} logs saved to recipes/pod-logs/"`

File: .github/workflows/functional-test.yaml [hunk 3]
```diff
@@ -559,6 +560,30 @@ jobs:
...
+          done
+          echo "Pod logs saved to recipes/pod-logs/"
+          # Get kubernetes events and save to file
+          kubectl get events -n $namespace > recipes/pod-logs/events.txt
```
Comment (ytimocin) [inline .github/workflows/functional-test.yaml:578]: And maybe another log after this line to state that the events are saved?

File: .github/workflows/functional-test.yaml (hunk 1 above)
Comment (sk593) [reply .github/workflows/functional-test.yaml:563]: No it wouldn't. We wanted to track recipe upload failures specifically, but we could just track the events whenever they're uploaded

File: .github/workflows/functional-test.yaml (hunk 2 above)
Comment (sk593) [reply .github/workflows/functional-test.yaml:574]: There's a line below that logs when all of them are uploaded but will update to this

**Pull request closed without merging**